tests/test_validate.py

```diff
@@ -107,3 +111,39 @@ def check(prurl):
     assert len(check(', https://flit.readthedocs.io/')) == 1
     # Name longer than 32 chars
     assert len(check('Supercalifragilisticexpialidocious, https://flit.readthedocs.io/')) == 1
+
+
+@pytest.fixture
+def patch_for_get_cache_dir():
+    # storing current environmental variables for resetting later
+    original_xdg = fv.os.environ.get("XDG_CACHE_HOME", None)
+    original_expanduser = fv.os.path.expanduser
+
+    # change the environmental variables for the test
+    fv.os.environ["XDG_CACHE_HOME"] = "/dev/null/nonexistent"
+    fv.os.path.expanduser = lambda x: "/dev/null/nonexistent"
```

Comment:
If you use the monkeypatch fixture, with its setattr & setenv methods, you don't need to do the cleanup manually. Which is good, because there's a minor bug - it won't clear XDG_CACHE_HOME if it was unset before.

https://docs.pytest.org/en/latest/monkeypatch.html

Reply:
Thanks for pointing out the monkeypatch feature from pytest – I should read the docs more often. I was considering using pytest.mock but didn't want to introduce a new dependency. Nice to know, that pytest has also a simple patch feature built in. 

I changed the fixture to use the monkeypatch version in the new commit.